### ISSUE TITLE
UTC field for every operation

### ISSUE THREAD

**Issue by orni137** [buildingSMART/BCF-API#186]
Could you please add a UTC for changing of every field in the header of issue?
It’s really difficult to build a good collaboration system if you don’t know the time of modification of each field.
It helps to merge events and to resolve event conflicts.

**Comment by GeorgDangl** [comment]
Hi @orni137,

we've got the `modified_date` property in comments and topics, which should be an ISO 8601 string which supports timezone offsets. Are there more resources you'd like to have this property on?

**Comment by orni137** [comment]
Hi Georg,

I meant the time for issue header fields. Especially when a user was offline and then is online again.
We can't understand when the issue header field was modified and who's made it last.

**Comment by GeorgDangl** [comment]
What do you mean with the header field? The [File Services](https://github.com/buildingSMART/BCF-API#43-file-services)?

**Comment by orni137** [comment]
I mean the fields in Topic header:
<xs:element name="Title" type="xs:string"/>
            <xs:attribute name="TopicStatus" type="TopicStatus"/>
            <xs:element name="Priority" type="Priority" minOccurs="0"/>
            <xs:element name="DueDate" type="xs:dateTime" minOccurs="0"/>
            <xs:element name="AssignedTo" type="UserIdType" minOccurs="0"/>
            <xs:element name="Stage" type="Stage" minOccurs="0"/>

**Comment by GeorgDangl** [comment]
There's actually a field available, it's called `ModifiedDate`, [you can find it here](https://github.com/buildingSMART/BCF-XML/blob/release_2_1/Schemas/markup.xsd#L69-L70).

```xml
<xs:element name="ModifiedDate" type="xs:dateTime" minOccurs="0"/>
<xs:element name="ModifiedAuthor" type="UserIdType" minOccurs="0"/>
```

Would that be what you need?

Please be aware that this is for the **BCF XML** format, which is it's own format. This repository, **BCF API**, is about a webservice implementation for the BCF standard. Both should have the same features, but they look a bit different.

**Comment by orni137** [comment]
Yes, unfortunately we still work with BCF XML. We know about that two fieldes:
<xs:element name="ModifiedDate" type="xs:dateTime" minOccurs="0"/>
<xs:element name="ModifiedAuthor" type="UserIdType" minOccurs="0"/>
But we need a ModifiedDate field for other 5 critical fields like this:

<xs:element name="TitleModifiedDate" type="xs:dateTime" minOccurs="0"/>
<xs:element name="TopicStatusModifiedDate" type="xs:dateTime" minOccurs="0"/>
etc.

**Comment by GeorgDangl** [comment]
Hey @orni137, we've talked about this proposal in the BCF Hackathon this week.
We agree with you that there's no built-in way to see when single properties have been changed, but this should be handled by the importing applications themselves. The `ModifiedDate` on the topic itself should be enough to check if a topic has been changed or not, then it's easy to figure out what has been changed.

In the API, this is less of a problem, since many parts of an issue have their own endpoints and representations.
